OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -81,14 +81,17 @@ public virtual void Tick(Actor self)
 			else
 				realignTick = 0;
 
-			var df = DesiredFacing ?? (facing != null ? facing.Facing : TurretFacing);
-			TurretFacing = Util.TickFacing(TurretFacing, df, info.ROT);
+			self.World.AddFrameEndTask(w =>
```

Comment:
This does mostly fix the problem, but not for the reason you think (or at least explained).  AttackTurreted Requires<> Turreted, so the order that these traits are constructed and tick is well defined (Turreted first, then AttackTurreted).

The problem is that, because Turreted has already ticked, any desired targets set by AttackTurreted cannot be acted on until the following tick.  Adding the FrameEndTask defers the the turret turning until after any other traits have set FaceTarget.  This combined with the pre-emptively ticked turret facing from FaceTarget removes this one-tick delay between turning and firing.

This exposes a new problem: anything that queries the turret facing between calling FaceTarget and the FrameEndTask will retrieve the wrong value, which may cause new regressions.  This will include at least the Armament / firing code, but probably also others.